Firebase/Messaging/FIRMessaging.m

```diff
@@ -184,24 +197,36 @@ - (void)dealloc {
 #pragma mark - Config
 
 + (void)load {
-  [[NSNotificationCenter defaultCenter] addObserver:self
-                                           selector:@selector(didReceiveConfigureSDKNotification:)
-                                               name:kFIRAppReadyToConfigureSDKNotification
-                                             object:nil];
+  [FIRApp registerAsConfigurable:self];
+  [FIRComponentContainer registerAsComponentRegistrant:self];
```

Comment:
Not familiar with the new init logic for core. Just want to confirm this would be the way of handling it from now on.